sequencer/src/block2.rs

```diff
@@ -647,28 +727,31 @@ mod test {
                 );
 
                 // test ns proof
-                let (ns_payload_flat_from_proof, ns_proof) = block
-                    .namespace_with_proof(&actual_ns_table, ns_idx)
+                let ns_proof2 = block
+                    .namespace_with_proof(
+                        &actual_ns_table,
+                        ns_idx,
+                        &vid,
+                        disperse_data.common.clone(),
+                    )
                     .unwrap();
                 assert_eq!(
-                    ns_payload_flat_from_proof, derived_ns.payload_flat,
+                    ns_proof2.ns_payload_flat, derived_ns.payload_flat,
                     "namespace {} incorrect payload bytes returned from namespace_with_proof",
                     ns_id.0,
                 );
-                // NOTE: There is no NamespaceProof::verify method because it's quite simple.
-                // compare: there is a TxInclusionProof::verify method for txs because that's complex.
-                // TODO make a NamespaceProof::verify method?
-                vid.payload_verify(
-                    Statement {
-                        payload_subslice: &ns_payload_flat_from_proof,
-                        range: actual_ns_payload_range,
-                        commit: &disperse_data.commit,
-                        common: &disperse_data.common,
-                    },
-                    &ns_proof,
-                )
-                .unwrap()
-                .unwrap_or_else(|_| panic!("namespace {} proof verification failure", ns_id.0));
+                let (ns_proof2_txs, ns_proof2_ns_id) = ns_proof2
```

Comment:
I guess there was some discussion about this API, but it seems unnatural for a proof verification to return a list of txs and a namespace id instead of a `boolean / Result<(),Error>`

Reply:
Yeah, there was discussion offline. I'm told that it makes life easy for use as an RPC endpoint. I don't ask questions. 😉